src/plugins/data/public/search/session/sessions_mgmt/components/table/table.tsx

```diff
@@ -63,30 +63,41 @@ export function SearchSessionsMgmtTable({
 
   // refresh behavior
   const doRefresh = useCallback(async () => {
+    if (refreshTimeoutRef.current) {
+      clearTimeout(refreshTimeoutRef.current);
+      refreshTimeoutRef.current = null;
+    }
+
     setIsLoading(true);
     const renderResults = (results: UISession[]) => {
       setTableData(results);
     };
     showLatestResultsHandler.current = renderResults;
-    let results: UISession[] = [];
-    try {
-      results = await api.fetchTableData();
-    } catch (e) {} // eslint-disable-line no-empty
+
+    if (document.visibilityState !== 'hidden') {
+      let results: UISession[] = [];
+      try {
+        results = await api.fetchTableData();
+      } catch (e) {} // eslint-disable-line no-empty
+
+      if (showLatestResultsHandler.current === renderResults) {
+        renderResults(results);
+        setIsLoading(false);
+      }
+    }
 
     if (showLatestResultsHandler.current === renderResults) {
-      renderResults(results);
-      setIsLoading(false);
+      if (refreshTimeoutRef.current) clearTimeout(refreshTimeoutRef.current);
+      refreshTimeoutRef.current = window.setTimeout(doRefresh, refreshInterval);
     }
-  }, [api]);
+  }, [api, refreshInterval]);
 
   // initial data load
   useEffect(() => {
     doRefresh();
     searchUsageCollector.trackSessionsListLoaded();
   }, [doRefresh, searchUsageCollector]);
 
-  useInterval(doRefresh, refreshInterval);
```

Comment:
These are small improvements to fetching on management screen because find() call is expensive now: 

- useInterval was refactored to a setTimeout, so that we had control of scheduling a new timeout only after on-going refresh completes. 

- Also added a shortcut `if (document.visibilityState !== 'hidden') {` to skip refresh in case page is not visible. 

- Reduced default fetched session count from 10000 to 100

These were easy attempts to reduce fetching on session screen (because the call is more expensive now), but we should discuss more impactful improvements: pagination and removing auto-refresh. I plan to bring up this to a search sync, but I suggest we merge this for now.